ui/tests/integration/components/confirm-action-test.js

```diff
@@ -0,0 +1,45 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module('Integration | Component | confirm-action', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders and on click shows the correct icon', async function (assert) {
+    let confirmAction = sinon.spy();
+    this.set('onConfirm', confirmAction);
+    await render(hbs`
+      <ConfirmAction
+        @onConfirmAction={{confirmAction}}
```

Comment:
Not sure if the component would throw an error if you add the assertion for the chevron changing direction on cancel, but this may have to change to: 
```suggestion
        @onConfirmAction={{this.onConfirm}}
```

Reply:
Looking at some other instances of using sinon.spy inside the test itself the const without the this is the correct approach. Modified the other usage so they're both consistent.

Reply:
But yes, I did have the name wrong, so it's onConfirm.

Reply:
Gotcha! Thanks for clearing it up